### DISCUSSION TITLE
AR with Hand Tracking

### DISCUSSION THREAD

**Discussion by VReamsports** [google/model-viewer#2236] [Q&A]
Hi I want to make an AR demo for myself with using hand tracking. The project in my mind is like the Iron Man Hand cannon as 3D object on my hand and tracking it. How can I do that ? someone can explain or help me with it ? ty.

**Comment by elalish** [comment]
For that you'll need a lower-level API than model-viewer. Here is some related work, maybe you can ask on these threads? https://twitter.com/pushmatrix/status/1237373570277191687

Someday WebXR may get image tracking and then we might enable something like this in model-viewer. Perhaps have it track a tattoo on your hand?

**Comment by VReamsports** [comment]
> For that you'll need a lower-level API than model-viewer. Here is some related work, maybe you can ask on these threads? https://twitter.com/pushmatrix/status/1237373570277191687
> 
> Someday WebXR may get image tracking and then we might enable something like this in model-viewer. Perhaps have it track a tattoo on your hand?

That would be useful but I'm not planning to work on VR, I need AR just like model viewer to track my hand and put 3D object on it I guess. There is a js called handtrack.js it's tracking your hand no matter what so I just modified it some to track my wraist too but I don't know how to integrate 3D model to my hand or something. So I thought model viewer maybe can add model to my tracked hand.

**Comment by elalish** [comment]
Not yet, though it's a possible future addition. Meanwhile you'll probably need three.js (what we're built on top of).
